packages/server/utils/analytics/analytics.ts

```diff
@@ -0,0 +1,101 @@
+import Meeting from '../../database/types/Meeting'
+import MeetingMember from '../../database/types/MeetingMember'
+import MeetingTemplate from '../../database/types/MeetingTemplate'
+import {TeamPromptResponse} from '../../postgres/queries/getTeamPromptResponsesByIds'
+import segment from '../segmentIo'
+import {createMeetingTemplateAnalyticsParams} from './helpers'
+import {SegmentAnalytics} from './segment/SegmentAnalytics'
+
+export type AnalyticsEvent = 'Meeting Started' | 'Meeting Joined' | 'Meeting Completed'
+
+/**
+ * Provides a unified inteface for sending all the analytics events
+ */
+class Analytics {
+  private segmentAnalytics: SegmentAnalytics
+
+  constructor() {
+    this.segmentAnalytics = new SegmentAnalytics(segment)
+  }
+
+  teamPromptEnd = (
+    completedMeeting: Meeting,
+    meetingMembers: MeetingMember[],
+    responses: TeamPromptResponse[]
+  ) => {
+    const userIdsResponses: Record<string, string> = responses.reduce(
+      (previous, response) => ({...previous, [response.userId]: response.plaintextContent}),
+      {}
+    )
+    meetingMembers.forEach((meetingMember) => {
+      const plaintextResponseContent = userIdsResponses[meetingMember.userId]
+      this.meetingEnd(meetingMember.userId, completedMeeting, meetingMembers, {
+        responseAdded: plaintextResponseContent && plaintextResponseContent.length > 0
```

Comment:
-1 This will not always resolve to a boolean value, e.g., when `plaintextResponseContent` is empty

Reply:
oh, good catch! fixed